src/BaGet.Web/Controllers/Registration/RegistrationIndexController.cs

```diff
@@ -1,171 +1,223 @@
+                Published = package.Published;
+                RequireLicenseAcceptance = package.RequireLicenseAcceptance;
+                Summary = package.Summary;
+                Tags = package.Tags;
+                Title = package.Title;
+                DependencyGroups = ToDependencyItems(package);
+            }
+          
+            private DependencyGroupItem[] ToDependencyItems(Package package)
+            {
+                List<DependencyGroupItem> groups = new List<DependencyGroupItem>();
+
+                var targetFrameworks = package.Dependencies.Select(a => a.TargetFramework).Distinct();
+                foreach(var tf in targetFrameworks)
+                {
+                    var depGroupId = $"https://api.nuget.org/v3/catalog0/data/2015.02.01.06.24.15/{package.Id}.{package.Version}.json#dependencygroup/{tf}";
+                    var groupItem = new DependencyGroupItem
+                    {
+                        Id=depGroupId,
+                        Type= "PackageDependencyGroup",
+                        TargetFramework=tf
+
+                    };
+
+                    foreach(var dep in package.Dependencies.Where(a => a.TargetFramework == tf)){
+                        var depItem = new DependencyItem
+                        {
+                            DepId= depGroupId+"/"+dep.Id,
+                            Type= "PackageDependency",
+                            Id=dep.Id,
+                            Range=dep.VersionRange
+                        };
+                        groupItem.Dependencies.Add(depItem);
+                    }
+
+                    groups.Add(groupItem);
+                }
+                return groups.Count>0?groups.ToArray():null;
+            }
+
+            [JsonProperty(PropertyName = "@id")]
+            public string CatalogUri { get; }
+
+            [JsonProperty(PropertyName = "id")]
+            public string PackageId { get; }
+
+            public string Version { get; }
+            public string Authors { get; }
+            public string Description { get; }
+            public long Downloads { get; }
+            public bool HasReadme { get; }
+            public string IconUrl { get; }
+            public string Language { get; }
+            public string LicenseUrl { get; }
+            public bool Listed { get; }
+            public string MinClientVersion { get; }
+            public string PackageContent { get; }
+            public string ProjectUrl { get; }
+            public string RepositoryUrl { get; }
+            public string RepositoryType { get; }
+            public DateTime Published { get; }
+            public bool RequireLicenseAcceptance { get; }
+            public string Summary { get; }
+            public string[] Tags { get; }
+            public string Title { get; }
+            public DependencyGroupItem[] DependencyGroups { get; }
+        }
+        private class DependencyGroupItem
```

Comment:
>DependencyGroupItem [](start = 22, length = 19)

`CatalogEntry` is intended to be immutable. Can you add a constructor for `DependencyGroupItem` and `DependencyItem` and remove all `set`s on properties?